File: EASSET.Web/Modules/ServerTypes/MasterData/UPBForm.ts
import { PrefixedContext, initFormType } from "@serenity-is/corelib";
import { ServiceLookupEditor, StringEditor } from "serenity.corelib";

export interface UPBForm {
    UnitId: ServiceLookupEditor;
    SubUnitId: ServiceLookupEditor;
    UpbKode: StringEditor;
    UpbNama: StringEditor;
}

export class UPBForm extends PrefixedContext {
    static readonly formKey = 'MasterData.UPB';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!UPBForm.init)  {
            UPBForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;

            initFormType(UPBForm, [
                'UnitId', w0,
                'SubUnitId', w0,
                'UpbKode', w1,
                'UpbNama', w1
            ]);
        }
    }
}

File: EASSET.Web/Modules/MasterData/UPB/UPBForm.cs
using EASSET.Helpers;
using Org.BouncyCastle.Bcpg.OpenPgp;
using Serenity.ComponentModel;
using StackExchange.Exceptional.Internal;
using System;
using System.Data;
using System.Windows.Markup;

namespace EASSET.MasterData.Forms;

[FormScript("MasterData.UPB")]
[BasedOnRow(typeof(UPBRow), CheckNames = true)]

public class UPBForm
{
  
    //[DefaultValue(), ReadOnly(true)]
    //public string ThAng { get; set; }
    public string UnitId { get; set; }
    public string SubUnitId { get; set; }
    public string UpbKode { get; set; }
    public string UpbNama { get; set; }

    //public string Op { get; set; }
    //public DateTime Lu { get; set; }
    //public string Pc { get; set; }
    //public short Dlt { get; set; }
}